src/async_wrap.h

```diff
@@ -152,7 +152,7 @@ class AsyncWrap : public BaseObject {
   static void EmitAfter(Environment* env, double async_id);
   static void EmitPromiseResolve(Environment* env, double async_id);
 
-  void EmitDestroy();
+  void EmitDestroy(bool from_gc = false);
```

Comment:
non-blocking nit... would prefer an enum here rather than a bool but that can be done in a separate PR

Reply:
I generally agree with that but here there’s effectively only one call site where passing `true` ever makes sense, so I don’t think this really falls under “boolean trap”.